tests/test_filesystem.py

```diff
@@ -0,0 +1,83 @@
+import pytest
+from prefect.testing.utilities import AsyncMock
+
+import prefect_github
+from prefect_github import GitHubCredentials
+from prefect_github.filesystem import GitHubRepository
+
+
+class TestGitHub:
+    async def test_subprocess_errors_are_surfaced(self):
```

Comment:
Can we test the validator `_ensure_credentials_go_with_https` and also `create_repo_url`?

Reply:
I want to gather a couple of opinions on this one in Slack. I've been going through a book that has some strong opinions on test design and sticking to testing public methods when possible.